middleware/legacy/rules/search.js

```diff
@@ -0,0 +1,133 @@
+// Redirect legacy search query parameters
+
+import qs from 'qs';
+
+import { unquotableFacets } from '../../../plugins/europeana/search';
+
+function mapCollectionToTheme(collection) {
+  const map = {
+    'world-war-I': 'ww1',
+    'industrial-heritage': 'industrial',
+    'manuscripts': 'manuscript',
+    'maps': 'map',
+    'natural-history': 'nature',
+    'newspapers': 'newspaper'
+  };
+  return map[collection] || collection;
+}
+
+const queryFacetParameterMappings = {
+  default(query, key, value) {
+    for (const filterValue of [].concat(value)) {
+      const queryFilterValue = unquotableFacets.includes(key) ? filterValue : `"${filterValue}"`;
+      query.qf.push(`${key}:${queryFilterValue}`);
+    }
+  },
+  'REUSABILITY'(query, key, value) {
+    query.reusability = [].concat(value).join(',');
+  },
+  api(query, key, value) {
+    const apiFilter = [].concat(value)[0];
+    if (apiFilter === 'default') {
+      query.api = 'metadata';
+    } else if (apiFilter === 'api') {
+      query.api = 'fulltext';
+    }
+  }
+};
+
+const queryParameterMappings = {
+  default(query, key, value) {
+    if (Array.isArray(query[key])) {
+      query[key] = query[key].concat([].concat(value));
+    } else {
+      query[key] = value;
+    }
+  },
+  page(query) {
+    delete query.page;
+  },
+  per_page(query) { // eslint-disable-line camelcase
+    delete query.per_page;
+  },
+  q(query, key, value) {
+    query.query.unshift(value);
+  },
+  qf(query, key, value) {
+    query.query = query.query.concat([].concat(value));
+  },
+  f(query, key, value) {
+    for (const field in value) {
+      if (queryFacetParameterMappings[field]) {
+        queryFacetParameterMappings[field](query, field, value[field]);
+      } else {
+        queryFacetParameterMappings.default(query, field, value[field]);
+      }
+    }
+  },
+  range(query, key, value) {
+    for (const field in value) {
+      const rangeBegin = value[field]['begin'];
+      const rangeEnd = value[field]['end'];
+      if (rangeBegin === rangeEnd) {
+        query.qf.push(`${field}:${rangeBegin}`);
```

Comment:
these need to account for empty strings and add '*'

Reply:
Indeed. That's done, with unit tests.